command/connect/envoy/envoy.go

```diff
@@ -507,8 +518,15 @@ func (c *cmd) templateArgs() (*BootstrapTplArgs, error) {
 		adminAccessLogPath = DefaultAdminAccessLogPath
 	}
 
+	// Fallback to the old certificate configuration, if none was defined.
+	if xdsAddr.AgentTLS && c.grpcCAFile == "" {
+		c.grpcCAFile = httpCfg.TLSConfig.CAFile
+	}
+	if xdsAddr.AgentTLS && c.grpcCAPath == "" {
+		c.grpcCAPath = httpCfg.TLSConfig.CAPath
+	}
```

Comment:
Is this the correct behavior? It's likely that any customers relying on the old CA cert configuration will break if I don't add in this extra behavior.

Reply:
so the gist here is "if i am already sure i should use TLS for xds, THEN fall back on the unprefixed flag/envvar place to get the CA stuff from"? If so then that sounds right.